### ISSUE TITLE
Calico-node high cpu Usage

### ISSUE THREAD

**Issue by cloud-66** [projectcalico/calico#3820]
In my kubernetes cluster there're  high cpu usage by calico-node pods . approximately 200m-400m cpu usage on every node.
Overal, pods in kube-system namespace comsume 10 cpu

## Expected Behavior
Less cpu usage by calico
## Current Behavior
calico use 10 cpu in 34 node cluster

## Possible Solution
May be I need to trace cpu usage, but i don't know how to do this.

## Steps to Reproduce (for bugs)
Calico used in cluster with  34 nodes , more than 4138 pod (not all running on the same time) and 2490 service
ipvs mode used

## Your Environment
Calico v3.11.2  https://docs.projectcalico.org/v3.11/manifests/calico.yaml
kubernetes v1.17.2
Ubuntu 18.04
ipvs

**Comment by cloud-66** [comment]
cpu usage from metrics-server

calico-kube-controllers-7ff6cb4c78-9bmjl        1m           20Mi
calico-node-28npp                               301m         88Mi
calico-node-2dfb5                               189m         82Mi
calico-node-2v9fs                               201m         91Mi
calico-node-5kmvv                               411m         80Mi
calico-node-5pjkb                               314m         92Mi
calico-node-7tfkt                               60m          72Mi
calico-node-b4xps                               277m         81Mi
calico-node-bgqcw                               241m         79Mi
calico-node-cdzzn                               303m         80Mi
calico-node-d2mps                               298m         89Mi
calico-node-f2pbx                               158m         92Mi
calico-node-fnrhn                               456m         81Mi
calico-node-g7286                               332m         81Mi
calico-node-gv79v                               417m         89Mi
calico-node-jfx82                               238m         82Mi
calico-node-jjx8w                               135m         82Mi
calico-node-k72km                               189m         86Mi
calico-node-lprx6                               249m         82Mi
calico-node-m7swn                               200m         90Mi
calico-node-nvfw4                               329m         80Mi
calico-node-pmm9j                               292m         86Mi
calico-node-pnr67                               265m         82Mi
calico-node-qf6vg                               76m          81Mi
calico-node-qjl4x                               110m         86Mi
calico-node-qnmxd                               370m         81Mi
calico-node-r2vvb                               228m         81Mi
calico-node-rqmhn                               223m         94Mi
calico-node-srzng                               82m          87Mi
calico-node-w9zs8                               389m         87Mi
calico-node-wfmwr                               384m         86Mi
calico-node-wvhbj                               54m          87Mi
calico-node-wzmkt                               306m         85Mi
calico-node-x4ths                               239m         83Mi
calico-node-zd8l4                               302m         81Mi

**Comment by ToughLama** [comment]
Please show your calico-node config!!! 
exec => kubectl -n kube-system get daemonset calico-node -oyaml

**Comment by caseydavenport** [comment]
@song-jiang @fasaxc have we done any profiling of felix in IPVS mode? Do we know if this is expected? 

At a glance, it looks like you're running on average about 120 pods per Kubernetes node, so a fairly densely packed cluster. calico/node needs to do more work as the number of pods in the cluster and the number of local pods on that node grows.

I'd recommend trying to see exactly what process(es) are using CPU within calico/node. Could you try checking wtihin a few pods to see which process is consuming the most?

**Comment by fasaxc** [comment]
@cloud-66 how much pod churn is there on your cluster?  Calico does more work when pods are churned.  To capture a CPU dump:
```
$ kubectl exec -n kube-system calico-node-XXXX sv 2 felix 
$ kubectl exec -n kube-system calico-node-XXXX ls /tmp
$ kubectl exec -n kube-system calico-node-j7s69 ls /tmp
felix-cpu-<timestamp>.pprof
$ kubectl exec -n kube-system calico-node-j7s69 cat /tmp/felix-cpu-<timestamp>.pprof > ./felix-cpu-<timestamp>.pprof
```
If you upload the pprof file to this issue I can help unpick it.

**Comment by cloud-66** [comment]
Hello. Sorry I didn't answer for a long time. I had an incident in the cluster a week ago.I don't think he was caused by calico. But because I had some errors in dmesg about ipvs (i attached old screen, but i had with this errors few weeks), and high cpu usage, I decided to switch to iptables mode. During recovering cluster after a failure i swithed to iptables and restart all nodes for cleaning routing rules
![изображение](https://user-images.githubusercontent.com/41143306/89638212-1702e500-d8b4-11ea-9281-ec12db8d102e.png)

Now, in iptables mode , i see smaller CPU usage
![Screenshot_5](https://user-images.githubusercontent.com/41143306/89641828-394c3100-d8bb-11ea-98aa-3c8e7dbd55aa.png)
